### PR TITLE
zklogin: Use ZkLoginAuthenticatorAsBytes for serde

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

Use ZkLoginAuthenticatorAsBytes instead in CompressedSignature. Added multisig zklogin tests with various test cases. 
Followup to: https://github.com/MystenLabs/sui/pull/14145

## Test Plan 

unit test and existing tests (consistency test in typescript still passes)

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/multisig.rs [hunk 1]
```diff
@@ -291,22 +293,6 @@ impl MultiSig {
...
         }
         self.multisig_pk.validate()?;
-        let sigs = std::mem::take(&mut self.sigs);
-        let mut new_sigs = Vec::with_capacity(sigs.len());
-
-        for s in sigs {
-            match s {
-                CompressedSignature::ZkLogin(mut s) => match s {
-                    GenericSignature::ZkLoginAuthenticator(ref mut z) => {
-                        z.inputs.init()?;
```
Comment (joyqvq) [inline crates/sui-types/src/multisig.rs:301]: no longer needed since init is already handled on line 185

File: crates/sui-types/src/multisig.rs [hunk 2]
```diff
@@ -344,8 +335,7 @@ impl ToFromBytes for MultiSig {
         }
         let mut multisig: MultiSig =
             bcs::from_bytes(&bytes[1..]).map_err(|_| FastCryptoError::InvalidSignature)?;
```
Comment (aschran) [inline crates/sui-types/src/multisig.rs:337]: I think this line still needs to be removed somehow as it's an anti-pattern (per @bmwill)

Either that or MultiSig::from_bytes must not be called in any impl of Deserialize

File: crates/sui-types/src/unit_tests/multisig_tests.rs
Comment (aschran) [inline crates/sui-types/src/unit_tests/multisig_tests.rs:1]: can this PR also include the original issue-repro test from @tzakian?

**Review by aschran — CHANGES_REQUESTED** [review]

File: crates/sui-types/src/unit_tests/multisig_tests.rs
Comment (joyqvq) [reply crates/sui-types/src/unit_tests/multisig_tests.rs:1]: its not producible with the current layout, there is a validation test on line 1009 in multisig_tests.rs

File: crates/sui-types/src/multisig.rs (hunk 2 above)
Comment (joyqvq) [reply crates/sui-types/src/multisig.rs:337]: not sure what the alternative should be, whats the recommended pattern?

also can be addressed in a follow up as this is used a few times in other places, not directly related to this pr.

**Review by aschran — APPROVED** [review]

**Comment by joyqvq** [comment]
as @kchalkias we will put this behind feature flag, currently disabled for mainnet

File: crates/sui-types/src/multisig.rs (hunk 2 above)
Comment (bmwill) [reply crates/sui-types/src/multisig.rs:337]: Correct, all uses of bcs from within a Deserialize impl should be removed

**Pull request merged**